### ISSUE TITLE
name argument in attachment not working

### ISSUE THREAD

**Issue by nlooije** [datawookie/emayili#91]
It seems like the name argument for attachment is not working as intended (or as i understand it).
Below is a minimal example; a jpeg is created and saved to a tempfile path and added as attachment.
The name 'test.jpg' is passed to the function call for attachment:
```
path <- tempfile(fileext='.jpg')

jpeg(path)
plot(1:10)

envelope() %>%
    from('xxx') %>%
    to('yyy') %>%
    subject('Test') %>%
    attachment(path=path, name='test.jpg', type='image/jpeg') %>%
    smtp(verbose = TRUE)
```
But the filename for the attached file is the base filename of the temp file (as is apparantly the default if no name argument is given according to the documentation)
```
Date:                      Thu, 14 Oct 2021 10:07:34 GMT
X-Mailer:                  {emayili}-0.6.3
MIME-Version:              1.0
From:                      xxx
To:                        yyy
Subject:                   Test
Content-Type:              image/jpeg; name="test.jpg"
Content-Disposition:       attachment; filename="file619d483db7c2.jpg"
Content-Transfer-Encoding: base64
X-Attachment-Id:           649571be
Content-ID:                <649571be>
Content-MD5:               1B2M2Y8AsgTpgAmY7PhCfg==
Uploaded 575 bytes... all done!
```

**Comment by datawookie** [comment]
Thanks, @nlooije. I'll take a look at this and get back to you. Might only be tomorrow morning though, sorry!

**Comment by datawookie** [comment]
Hi @nlooije, thanks again. This is fixed in 83e15d231d856beab9c091048f01f3096048b8c1. Please install from GitHub to get the update. It will go into the next release on CRAN. Best regards, Andrew.

```r
remotes::install_github("datawookie/emayili", ref = "83e15d231d856beab9c091048f01f3096048b8c1")
```

```
> library(emayili)
> path <- tempfile(fileext='.jpg')
> jpeg(path)
> plot(1:10)
# -----------------------------------------------------------------------------
# Using default file name.
# -----------------------------------------------------------------------------
> envelope() %>%
+   subject('Test') %>%
+   attachment(path) %>%
+   as.character() %>% cat()
Date:                      Fri, 15 Oct 2021 04:06:15 GMT
X-Mailer:                  {emayili}-0.6.5
MIME-Version:              1.0
Subject:                   Test
Content-Type:              image/jpeg; name="file82dac6c3f88f6.jpg"
Content-Disposition:       attachment; filename="file82dac6c3f88f6.jpg"
Content-Transfer-Encoding: base64
X-Attachment-Id:           52c12667
Content-ID:                <52c12667>
Content-MD5:               1B2M2Y8AsgTpgAmY7PhCfg==
# -----------------------------------------------------------------------------
# Using specific file name.
# -----------------------------------------------------------------------------
> envelope() %>%
+   subject('Test') %>%
+   attachment(path=path, name='test.jpg', type='image/jpeg') %>%
+   as.character() %>% cat()
Date:                      Fri, 15 Oct 2021 04:06:15 GMT
X-Mailer:                  {emayili}-0.6.5
MIME-Version:              1.0
Subject:                   Test
Content-Type:              image/jpeg; name="test.jpg"
Content-Disposition:       attachment; filename="test.jpg"
Content-Transfer-Encoding: base64
X-Attachment-Id:           bb18e992
Content-ID:                <bb18e992>
Content-MD5:               1B2M2Y8AsgTpgAmY7PhCfg==
```

**Comment by nlooije** [comment]
confirmed to be working, thx
